stdlib/dynarray.ml

```diff
@@ -202,130 +367,127 @@ let[@inline always] check_valid_length length arr =
 
    This precondition is typically guaranteed by knowing
    [0 <= i < length] and calling [check_valid_length length arr].*)
-let[@inline always] unsafe_get arr ~i ~length =
-  match Array.unsafe_get arr i with
-  | Empty -> Error.missing_element ~i ~length
-  | Elem {v} -> v
-
+let[@inline always] unsafe_get arr ~dummy ~i ~length =
+  let v = Array.unsafe_get arr i in
+  if Dummy.is_dummy v dummy
+  then Error.missing_element ~i ~length
+  else Dummy.unsafe_get v
 
 (** {1:dynarrays Dynamic arrays} *)
 
-let create () = {
-  length = 0;
-  arr = [| |];
-}
+let create () =
+  let Dummy.Fresh dummy = global_dummy in
+  Pack {
+    length = 0;
+    arr = [| |];
+    dummy = dummy;
+  }
 
 let make n x =
+  let Dummy.Fresh dummy = global_dummy in
+  let arr = Dummy.Array.make n x ~dummy in
   if n < 0 then Error.negative_length_requested "make" n;
```

Comment:
Why leave the length check if `Array.make` is already called ?
Did you mean to call it after the check ?

Reply:
I think so, indeed. Thanks!